client/my-sites/stats/stats-period-navigation/index.jsx

```diff
@@ -52,19 +52,36 @@ class StatsPeriodNavigation extends PureComponent {
 
 	handleArrowNext = () => {
 		const { date, moment, period, url, queryParams } = this.props;
+		const getQueryParams = new URLSearchParams( window.location.search );
+		const tabValue = getQueryParams.get( 'tab' );
 		const nextDay = moment( date ).add( 1, period ).format( 'YYYY-MM-DD' );
-		const nextDayQuery = qs.stringify( Object.assign( {}, queryParams, { startDate: nextDay } ), {
-			addQueryPrefix: true,
-		} );
+		const nextDayQuery = qs.stringify(
+			Object.assign(
+				{},
+				queryParams,
+				{ startDate: nextDay },
+				{ tab: tabValue ? tabValue : 'views' }
+			),
+			{
+				addQueryPrefix: true,
+			}
+		);
 		const href = `${ url }${ nextDayQuery }`;
 		this.handleArrowEvent( 'next', href );
 	};
 
 	handleArrowPrevious = () => {
 		const { date, moment, period, url, queryParams } = this.props;
 		const previousDay = moment( date ).subtract( 1, period ).format( 'YYYY-MM-DD' );
+		const getQueryParams = new URLSearchParams( window.location.search );
```

Comment:
I would recommend to assign `content.query`  injected by a `page.js` middleware to `queryParams` and use it here, otherwise it probably wouldn't work for Odyssey, because Odyssey uses hashes as for queries: pejTkB-2s#2-2-routing

Reply:
Great, thanks so much Jasper this is really helpful advice!

Reply:
Related code:

**Odyssey**
https://github.com/Automattic/wp-calypso/blob/b37ccf43981ced060e523616039b58834ea66188/apps/odyssey-stats/src/page-middleware/setup-context.jsx#L13

**Calypso**
https://github.com/Automattic/wp-calypso/blob/b37ccf43981ced060e523616039b58834ea66188/client/boot/common.js#L65

Pass it to top level components and then pass down to child component I suppose:
https://github.com/Automattic/wp-calypso/blob/b37ccf43981ced060e523616039b58834ea66188/client/my-sites/stats/controller.jsx#L368